src/main/java/com/marklogic/client/ext/schemasloader/impl/QbvDocumentFileProcessor.java

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.client.ext.schemasloader.impl;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.document.XMLDocumentManager;
+import com.marklogic.client.eval.ServerEvaluationCall;
+import com.marklogic.client.ext.file.DocumentFile;
+import com.marklogic.client.ext.file.DocumentFileProcessor;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.DOMHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import org.jdom2.Element;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.output.DOMOutputter;
+import org.springframework.util.FileCopyUtils;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QbvDocumentFileProcessor extends LoggingObject implements DocumentFileProcessor {
+
+	public static final String QBV_COLLECTION = "http://marklogic.com/xdmp/qbv";
+	public static final String QBV_XML_PLAN_NAMESPACE = "http://marklogic.com/plan";
+	public static final String QBV_XML_ROOT_ELEMENT = "query-based-view";
+	public static final String JAVASCRIPT_EVAL_TEMPLATE = "declareUpdate(); xdmp.invokeFunction(function() {'use strict'; const op = require('/MarkLogic/optic'); return %s }, {database: xdmp.database('%s')})";
+	public static final String XQUERY_EVAL_TEMPLATE = "xquery version \"1.0-ml\"; import module namespace op=\"http://marklogic.com/optic\" at \"/MarkLogic/optic.xqy\"; xdmp:invoke-function(function() {%s},<options xmlns=\"xdmp:eval\"><database>{xdmp:database('%s')}</database></options>)";
+
+	final private DatabaseClient databaseClient;
+	final private String qbvGeneratorDatabaseName;
+	final protected List<DocumentFile> qbvFiles = new ArrayList<>();
+	final private XMLDocumentManager docMgr;
+
+
+	/**
+	 * @param databaseClient - a MarkLogic DatabaseClient object for the Schemas database
+	 */
+	public QbvDocumentFileProcessor(DatabaseClient databaseClient, String qbvGeneratorDatabaseName) {
+		this.databaseClient = databaseClient;
+		this.qbvGeneratorDatabaseName = qbvGeneratorDatabaseName;
+		this.docMgr = databaseClient.newXMLDocumentManager();
+	}
+
+	@Override
+	public DocumentFile processDocumentFile(DocumentFile documentFile) {
+		if (isQbvQuery(documentFile)) {
+			// Defer processing until after all other schemas have been processed.
+			qbvFiles.add(documentFile);
+			return null;
+		} else {
+			return documentFile;
+		}
+	}
+
+	private boolean isQbvQuery(DocumentFile documentFile) {
+		String uri = documentFile.getUri();
+		String extension = documentFile.getFileExtension();
+		if (extension != null) {
+			extension = extension.toLowerCase();
+		}
+		return
```

Comment:
I added a "example.sjs" file to the qbv-schemas test directory, and that caused "test()" to fail. I think this needs to do the URI check first, and if that's true, then verify the extension.